components/admin_console/workspace-optimization/dashboard.tsx

```diff
@@ -137,29 +123,38 @@ const WorkspaceOptimizationDashboard = (props: Props) => {
         setTimeout(() => testSiteURL(onSuccess, onError, location.origin), 1000);
     }, []);
 
-    const dataKey = Object.keys(data);
+    const prepareDataForAccordion = () => {
+        const accData: any = [];
+        const dataKeys = Object.keys(data);
+
+        dataKeys.forEach((key: string) => {
+            const items = data[key].items.map((item) => (
+                <AccordionItem
+                    key={`${key}-item_${item.id}`}
+                    iconColor={getItemColor(item.status)}
+                >
+                    <h5><i className={classNames('icon', {'icon-check-circle-outline': item.status === 'none', 'icon-alert-outline': item.status === 'warning', 'icon-alert-circle-outline': item.status === 'error'})}/>{item.title}</h5>
+                    <p>{item.description}</p>
+                </AccordionItem>
+            ));
+            accData.push({
+                title: data[key].title,
+                description: data[key].description,
+                items,
+            });
+        });
+
+        return accData;
+    };
```

Comment:
we can extract this to the FC logic rather than have a function handling that.

Just use it like this:
```suggestion
    const accData: any = [];
    const dataKeys = Object.keys(data);

    dataKeys.forEach((key: string) => {
        const items = data[key].items.map((item) => (
            <AccordionItem
                key={`${key}-item_${item.id}`}
                iconColor={getItemColor(item.status)}
            >
                <h5><i className={classNames('icon', {'icon-check-circle-outline': item.status === 'none', 'icon-alert-outline': item.status === 'warning', 'icon-alert-circle-outline': item.status === 'error'})}/>{item.title}</h5>
                <p>{item.description}</p>
            </AccordionItem>
        ));
        accData.push({
            title: data[key].title,
            description: data[key].description,
            items,
        });
    });
```